src/item/enchantment/VanillaEnchantments.php

```diff
@@ -59,47 +60,181 @@ final class VanillaEnchantments{
-		self::register("RESPIRATION", new Enchantment(KnownTranslationFactory::enchantment_oxygen(), Rarity::RARE, ItemFlags::HEAD, ItemFlags::NONE, 3));
+		self::register("PROTECTION", new ProtectionEnchantment(KnownTranslationFactory::enchantment_protect_all(), Rarity::COMMON,
+			Flags::ARMOR,
+			Flags::NONE,
+			4,
+			0.75, null,
+			false,
+			fn($level) => 11 * ($level - 1) + 1,
+			fn($level, $minCost) => $minCost + 20
+		));
+		self::register("FIRE_PROTECTION", new ProtectionEnchantment(KnownTranslationFactory::enchantment_protect_fire(), Rarity::UNCOMMON,
+			Flags::ARMOR,
+			Flags::NONE,
+			4,
+			1.25,
+			[
+				EntityDamageEvent::CAUSE_FIRE,
+				EntityDamageEvent::CAUSE_FIRE_TICK,
+				EntityDamageEvent::CAUSE_LAVA
+				//TODO: check fireballs
+			], false,
+			fn($level) => 8 * ($level - 1) + 10,
+			fn($level, $minCost) => $minCost + 12
+		));
+		self::register("FEATHER_FALLING", new ProtectionEnchantment(KnownTranslationFactory::enchantment_protect_fall(), Rarity::UNCOMMON,
+			Flags::FEET,
+			Flags::NONE,
+			4,
+			2.5,
+			[
+				EntityDamageEvent::CAUSE_FALL
+			], false,
+			fn($level) => 6 * ($level - 1) + 5,
+			fn($level, $minCost) => $minCost + 10
+		));
+		self::register("BLAST_PROTECTION", new ProtectionEnchantment(KnownTranslationFactory::enchantment_protect_explosion(), Rarity::RARE,
+			Flags::ARMOR,
+			Flags::NONE,
+			4,
+			1.5,
+			[
+				EntityDamageEvent::CAUSE_BLOCK_EXPLOSION,
+				EntityDamageEvent::CAUSE_ENTITY_EXPLOSION
+			], false,
+			fn($level) => 8 * ($level - 1) + 5,
+			fn($level, $minCost) => $minCost + 12
+		));
+		self::register("PROJECTILE_PROTECTION", new ProtectionEnchantment(KnownTranslationFactory::enchantment_protect_projectile(), Rarity::UNCOMMON,
+			Flags::ARMOR,
+			Flags::NONE,
+			4,
+			1.5,
+			[
+				EntityDamageEvent::CAUSE_PROJECTILE
+			], false,
+			fn($level) => 6 * ($level - 1) + 3,
+			fn($level, $minCost) => $minCost + 15
+		));
+		self::register("THORNS", new Enchantment(KnownTranslationFactory::enchantment_thorns(), Rarity::MYTHIC,
+			Flags::TORSO,
+			Flags::HEAD | Flags::LEGS | Flags::FEET,
+			3, false,
+			fn($level) => 20 * ($level - 1) + 10,
+			fn($level, $minCost) => $minCost + 50
+		));
+		self::register("RESPIRATION", new Enchantment(KnownTranslationFactory::enchantment_oxygen(), Rarity::RARE,
+			Flags::HEAD,
+			Flags::NONE,
+			3, false,
+			fn($level) => 10 * $level,
+			fn($level, $minCost) => $minCost + 30
+		));
 
-		self::register("SHARPNESS", new SharpnessEnchantment(KnownTranslationFactory::enchantment_damage_all(), Rarity::COMMON, ItemFlags::SWORD, ItemFlags::AXE, 5));
-		//TODO: smite, bane of arthropods (these don't make sense now because their applicable mobs don't exist yet)
+		self::register("SHARPNESS", new SharpnessEnchantment(KnownTranslationFactory::enchantment_damage_all(), Rarity::COMMON,
+			Flags::SWORD | Flags::AXE,
```

Comment:
Sharpness cant be received from enchanting table for axes

Reply:
It can be. You can check in vanilla world and [here](https://minecraft.fandom.com/wiki/Enchanting#Summary_of_enchantments).